src/refinement/refinement.cpp

```diff
@@ -131,5 +130,19 @@ AmrTag FirstDerivative(MeshBlock *pmb, const ParArrayND<Real> &q,
   return AmrTag::same;
 }
 
+TaskStatus Tag(std::shared_ptr<MeshBlockData<Real>> &rc) {
+  auto pmb = rc->GetBlockPointer();
+  pmb->pmr->SetRefinement(CheckAllRefinement(rc));
+  return TaskStatus::complete;
+}
+
+TaskStatus Tag(std::shared_ptr<MeshData<Real>> &rc) {
+  for (int i = 0; i < rc->NumBlocks(); i++) {
+    auto &pbd = rc->GetBlockData(i);
+    auto status = Tag(pbd);
+  }
+  return TaskStatus::complete;
+}
+
```

Comment:
This is a nice solution. I'm not sure either is more performant, since we're still doing tagging per MeshBlock, but I don't know if there's a way around that.

Reply:
yeah agreed, it won't be more performant.  The `MeshData` interface is just for convenience for applications that are working mostly/exclusively with those objects.